squashfs/squashfs.go

```diff
@@ -321,3 +328,24 @@ func whichSearch(name string, paths []string) string {
 
 	return ""
 }
+
+func mksquashfsSupportsZstd() bool {
+	checkZstdSupported.Do(func() {
+		var stdoutBuffer strings.Builder
+		var stderrBuffer strings.Builder
+
+		cmd := exec.Command("mksquashfs", "--help")
+		cmd.Stdout = &stdoutBuffer
+		cmd.Stderr = &stderrBuffer
+
+		// Ignore errs here as `mksquashfs --help` exit status code is 1
```

Comment:
interesting, is this something common?

Reply:
More common than I'd like.  It's an easy trap to fall into when you're writing a utility like this and want to combine the '--help' flow with the more common in your code dispatches to 'malformed inputs so print usage and error exit'.  mkquashfs is not the only one doing this by any means.